chopper/lib/src/interceptor.dart

```diff
@@ -52,8 +52,10 @@ class HeadersInterceptor implements RequestInterceptor {
       applyHeaders(request, headers);
 }
 
-typedef FutureOr<Response> ResponseInterceptorFunc<Value>(
+typedef ResponseInterceptorFunc = FutureOr<Response> Function<Value>(
```

Comment:
I missed that !
We should force to not modifying the BodyType inside an interceptor by returning `FutureOr<Response<Value>>`

Reply:
Good point added it